### PR TITLE
Laravel 11.x Compatibility

### PR DESCRIPTION
**Pull request by laravel-shift**
This is an automated pull request from [Shift](https://laravelshift.com) to update your package code and dependencies to be compatible with Laravel 11.x.

**Before merging**, you need to:

- Checkout the `l11-compatibility` branch
- Review **all** comments for additional changes
- Thoroughly test your package

If you do find an issue, please report it by commenting on this PR to help improve future automation.

### CODE REVIEW

**Comment by laravel-shift** [comment]
:warning: The latest version of Laravel has a dependency for `phpunit/phpunit` of `^10.5`. If you would like to also upgrade your tests, you may run the [PHPUnit 10 Shift](https://laravelshift.com/upgrade-phpunit-10) for free.

**Comment by laravel-shift** [comment]
:warning: Shift detected a dependency for the `laravel/framework`. Unless you have a special requirement for depending on the entire Laravel framework, it's recommended to add dependencies for the specific `illuminate` packages instead.

**Comment by laravel-shift** [comment]
:warning: Shift detected GitHub Actions which run jobs using a version `matrix`. Shift attempted to [update your configuration](https://github.com/timacdonald/json-api/pull/49/commits/a329cd1fcea8e01ea84e1a269708416eb2bb211c?w=1) for Laravel 11. However, you should review these changes to ensure the desired combination of versions are built for your package.

**Comment by laravel-shift** [comment]
:alembic: **Using this package?** If you would like to help test these changes or believe them to be compatible, you may update your project to reference this branch.

To do so, temporarily add Shift's fork to the `repositories` property of your `composer.json`:

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/laravel-shift/json-api.git"
        }
    ]
}
```

Then update your dependency constraint to reference this branch:

```json
{
    "require": {
        "timacdonald/json-api": "dev-l11-compatibility",
    }
}
```

Finally, run: `composer update`

**Review by lloricode — CHANGES_REQUESTED** [review]
please add php8.3, thank you

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,128 +1,135 @@
...
+        with:
+          path: ${{ steps.composer-cache.outputs.dir }}
+          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+          restore-keys: ${{ runner.os }}-composer-
+
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '8.1'
```
Comment (lloricode) [inline .github/workflows/main.yml:40]: please use php 8.3 as latest version

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,128 +1,135 @@
...
+
+  tests:
+    runs-on: ubuntu-latest
+
+
+    name: PHP: ${{ matrix.php }}; Laravel: ${{ matrix.laravel }}; Prefer: ${{ matrix.prefer }}
+
+    strategy:
+      matrix:
+        php: ['8.1', '8.2']
```
Comment (lloricode) [inline .github/workflows/main.yml:63]: please add php8.3, thank you

**Review by lloricode — COMMENTED** [review]
can github action workflow separate files? it is difficult to track all workflows here

**Comment by lloricode** [comment]
will make PR as soon as I vack to my machine

File: .github/workflows/main.yml (hunk 1 above)
Comment (timacdonald) [reply .github/workflows/main.yml:33]: This one is okay.

File: .github/workflows/main.yml (hunk 2 above)
Comment (timacdonald) [reply .github/workflows/main.yml]: Updated, thanks!

Comment (lloricode) [reply .github/workflows/main.yml]: thanks :beers:

**Pull request merged**